core/src/main/java/org/mule/runtime/core/util/ExceptionUtils.java

```diff
@@ -275,4 +278,23 @@ public static Throwable getRootCauseException(Throwable exception) {
     return exception instanceof ErrorMessageAwareException ? ((ErrorMessageAwareException) exception).getRootCause() : exception;
   }
 
+  /**
+   * Propagate an exception through streams, may wrap it to avoid throwing from Reactor code.
+   *
+   * @param t throwable.
+   * @return possibly wrapped exception to avoid Reactor stopping the stream.
+   */
+  public static RuntimeException propagateReactor(Throwable t) {
+    return propagate(wrapReactor(t));
+  }
+
+  public static Throwable wrapReactor(Throwable t) {
```

Comment:
Javadoc. Also consider is there is a clearer name we can give this method, otherwise doesn't matter as long as javadoc is clear.